### ISSUE TITLE
Question about rounding

### ISSUE THREAD

**Issue by sergeyf** [mrtzh/Ladder.jl#1]
Howdy,

Thanks for the great paper and publishing your algorithm. I have a question about rounding.

For the fixed step size algorithm I see that [you round](https://github.com/mrtzh/Ladder.jl/blob/6640f98b86a0952492a5873e5efd9dd9367e5918/src/Ladder.jl#L65) as follows:

`l.current = round(loss,int(log(10,length(l.labels))))`

However, in the [parameter free ladder algorithm](https://github.com/mrtzh/Ladder.jl/blob/6640f98b86a0952492a5873e5efd9dd9367e5918/src/Ladder.jl#L55) there is no rounding:

`l.current = loss`

Can you explain why one has rounding (as well as its form) and the other does not?

I may have misunderstood how this is working if it's related to the notation of [R_s(f)]_alpha (not defined anywhere in the paper). Is this rounding or floor notation?

**Comment by mrtzh** [comment]
It's a bit arbitrary to be honest. The version with fixed step sizes corresponds to the algorithm in the paper that has provable guarantees. For those guarantees you need the rounding. Btw, the notation is explained in several places in the paper, e.g., below each algorithm figure.

The parameter free version doesn't have provable accuracy guarantees. It is largely meant to be a way to implement the algorithm heuristically in a way that seems to do well in practice. If you are in an adversarial setting, someone might be able to exploit the absence of rounding by sucking too much information out of the holdout set using the lower order digits of the answer. That's intuitively what the rounding prevents.

So, in short whether or not to round is a design choice that ought to depend on what environment the algorithm is supposed to perform in. If you want to be on the safe side, then rounding to say 6 digits of precision seems reasonable. The code you quoted should implement the function "round to log(n) bits of precision" where n is the size of the data set.

Does this answer your questions?

**Comment by sergeyf** [comment]
Yes, that answers it. 

Thank you very much!
